### PR TITLE
Add hypershift test image

### PR DESCRIPTION
**Pull request by enxebre**
Add hypershift test for autoscaler: For this it builds a new image "hypershift-test" which is the same than "hypershift-operator" but it uses openshift_release_golang-1.18 as the final image so it has make available.

Needs https://github.com/openshift/hypershift/pull/1441

First step for https://issues.redhat.com/browse/HOSTEDCP-483 and https://issues.redhat.com/browse/HOSTEDCP-482.

Once this is merge I'll add a job to run the image and tests in autoscaler repo.
Once this is settled I'll update the workflows to use the Makefile and everything else to use the workflows.
/hold

### CODE REVIEW

File: ci-operator/config/openshift/kubernetes-autoscaler/openshift-kubernetes-autoscaler-master.yaml [hunk 1]
```diff
@@ -31,6 +31,12 @@ promotion:
   name: "4.11"
   namespace: ocp
 releases:
+  hosted-ocp-previous:
```
Comment (alvaroaleman) [inline ci-operator/config/openshift/kubernetes-autoscaler/openshift-kubernetes-autoscaler-master.yaml:34]: Did you intend to add this release?

**Review by alvaroaleman — APPROVED** [review]

**Comment by alvaroaleman** [comment]
/assign @bparees 
Ben, can you have a look on the updated image mirroring config please?

**Comment by enxebre** [comment]
>/assign @bparees
Ben, can you have a look on the updated image mirroring config please?

fwiw is automatically generated.

Not sure the image fails to build https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29179/rehearse-29179-pull-ci-openshift-hypershift-main-images/1534840906555854848

```
Copying config sha256:ceda6843a091489f7c9646d41cb550e6cd428a7d903f4b82fdc85c5b1060ec9d
Writing manifest to image destination
Storing signatures
error: error copying source path /go/src/github.com/openshift/hypershift/. to .: exit status 1
```

**Comment by csrwng** [comment]
@enxebre because you don't specify a Dockerfile for the image, it defaults to the Dockerfile in the root of our repository. I think at the very least you need to specify an inline Dokerfile (see `dockerfile_literal` in https://docs.ci.openshift.org/docs/architecture/ci-operator/#building-container-images)

**Comment by bparees** [comment]
> @enxebre because you don't specify a Dockerfile for the image, it defaults to the Dockerfile in the root of our repository. 

i assumed that was deliberate, based on `For this it builds a new image "hypershift-test" which is the same than "hypershift-operator" `

**Comment by csrwng** [comment]
@enxebre do we still need this? 

/remove-lifecycle stale

**Comment by enxebre** [comment]
>@enxebre do we still need this?

We need ability to run e2e against other repos. I'll reopen when tacking this back.

**Pull request closed without merging**